### PR TITLE
cosiDriver: adds new API resource, runs code gen

### PR DESCRIPTION
**Pull request by krishchow**
This PR adds the COSIDriver resource discussed in the previous community meeting.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -3,3 +3,4 @@
 .build
 *.swp
 release-tools
+.idea
```
Comment (wlan0) [inline .gitignore:6]: ?

**Comment by wlan0** [comment]
The change looks good. Given today's discussion with @bswartz, I think we should hold off on this until we reach concensus on the approach

File: .gitignore (hunk 1 above)
Comment (krishchow) [reply .gitignore:6]: `.idea` is the folder created by JetBrains IDEs to store project metadata/configuration

**Pull request closed without merging**